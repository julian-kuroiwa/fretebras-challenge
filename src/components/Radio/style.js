import styled from 'styled-components';

export const Container = styled.div`
  &:not(:last-child) {
    margin-bottom: 10px;
  }

  input[type='radio'] {
    position: absolute;
    opacity: 0;

    & + .radio-label:before {
      content: '';
      background: #f4f4f4;
      border: 1px solid #b4b4b4;
      border-radius: 4px;
      display: inline-block;
      width: 20px;
      height: 20px;
      position: relative;
      top: -3px;
      margin-right: 10px;
      vertical-align: top;
      cursor: pointer;
      text-align: center;
      -webkit-transition: all 0.25s ease;
      transition: all 0.25s ease;
    }

    &:checked + .radio-label:before {
      background-color: #b4b4b4;
      box-shadow: inset 0 0 0 4px #f4f4f4;
    }
    &:focus + .radio-label:before {
      outline: none;
      border-color: #b4b4b4;
    }
    &:disabled + .radio-label:before {
      box-shadow: inset 0 0 0 4px #f4f4f4;
      border-color: #b4b4b4;
      background: #b4b4b4;
    }
    & + .radio-label:empty:before {
      margin-right: 0;
    }
  }
`;
