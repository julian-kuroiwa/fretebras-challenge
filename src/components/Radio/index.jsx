import React from 'react';

import { Container } from './style';

const Radio = ({ id, title, ...props }) => (
  <Container>
    <input id={id} value={id} type="radio" {...props} />
    <label htmlFor={id} className="radio-label">
      {title}
    </label>
  </Container>
);

export default Radio;
