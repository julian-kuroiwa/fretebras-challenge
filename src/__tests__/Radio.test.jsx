import { shallow, mount } from 'enzyme';
import React from 'react';

import Radio from '../components/Radio';

describe('Radio', () => {
  it('should match snapshot', () => {
    const component = shallow(<Radio />);

    expect(component).toMatchSnapshot();
  });

  it('should render with children', () => {
    const component = shallow(<Radio id="1" title="radio button" />);

    expect(component.find('label').text()).toBe('radio button');
  });

  it('should render with id', () => {
    const component = mount(<Radio id="1" title="radio button" />);

    expect(component.props().id).toBe('1');
  });
});
