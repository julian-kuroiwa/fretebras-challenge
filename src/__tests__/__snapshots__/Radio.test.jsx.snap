// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio should match snapshot 1`] = `
<styled.div>
  <input
    type="radio"
  />
  <label
    className="radio-label"
  />
</styled.div>
`;
